### PR TITLE
Rewrite EditProfileActivity in Kotlin

### LINKED ISSUE #504: Problems with Profile Editor's UI
There are several closely related problems:
- Right now, I can only remove my Bio by setting it to a whitespace (Mastodon strips whitespace from fields
- When I set my name to a whitespace, UI closes without errors, but name remains unchanged.

_Sent from my LEX820 using [FastHub-Libre](https://github.com/thermatk/FastHub-Libre/)_

### PR DESCRIPTION
**Pull request by connyduck**
- fixes #504 
- makes the activity actually work again lol
- saves the image in the app cache dir and uses proper multipart encoding
- fixes a bug in MainActivity where profiles would duplicate
- probably fixes the TransactionTooLarge crashes I am seeing in the Play console, because the base64 shenanigans are gone
- small layout improvements
- Kotlin yeah yeah

the random filenames seems to be necessary so mastodon accepts the images

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+        }
+
+        val header = if(headerChanged) {
+            val headerBody = RequestBody.create(MediaType.parse("image/png"), getCacheFileForName(HEADER_FILE_NAME))
+            MultipartBody.Part.createFormData("header", getFileName(), headerBody)
+        } else {
+            null
+        }
+
+        if(displayName == null && note == null && avatar == null && header == null) {
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt]: Not clear why this check is needed. Perhaps a comment would help.

File: app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+            EditProfileActivity.PickType.HEADER -> {
+                width = HEADER_WIDTH
+                height = HEADER_HEIGHT
+                cacheFile = getCacheFileForName(HEADER_FILE_NAME)
+            }
+            else -> {
+                throw AssertionError("PickType not set.")
+            }
+        }
+        ResizeImageTask(contentResolver, width, height, cacheFile, object : ResizeImageTask.Listener {
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt]: Of course it's not perfect that it captures activity but it shouldn't be too slow so I'm not sure if it's worth doing something with it.

File: app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+
+        internal interface Listener {
+            fun onSuccess(resizedImage: Bitmap?)
+            fun onFailure()
+        }
+    }
+
+}
+
+fun saveBitmapToFile(bitmap: Bitmap, file: File): Boolean {
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt]: It's kinda weird that it's outside the class but inside the file.

File: app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt (hunk 1 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt]: If everything is null, nothing has changed and there is no need to make a network request :)
Will add a comment.

File: app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt (hunk 2 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt]: I am pretty sure that this AsyncTask does not have a reference to the activity. Thats the reason why the function below is outside the class, the function is not accessible if it is a member of the class (it doesnt compile).

**Pull request merged**